.github/workflows/nightly-rebase-upstream.yml

```diff
@@ -0,0 +1,70 @@
+# Every night, rebase the Julia branches on top of raicode `master`.
+name: NightlyRebaseUpstream
+on:
+  schedule:
+    # every night at 11 PM, raicode nightly rebase runs at midnight
+    - cron: 0 23 * * *
+  workflow_dispatch:
+permissions:
+  contents: write
+  pull-requests: write
+jobs:
+  NightlyRebaseUpstream:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false  # run all jobs in the matrix even if one fails
+      matrix:
+        julia_branch:
+          - { name: "backports-release-1.9+RAI", upstream: "backports-release-1.9"}
+          - { name: "backports-release-1.10+RAI", upstream: "backports-release-1.10"}
+          - { name: "master+RAI", upstream: "master"}
+    steps:
+      - name: Checkout the repo and our branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ matrix.julia_branch.name }}
+          # Needed to make rebasing work (https://github.com/actions/checkout/issues/372)
+          fetch-depth: 0
+      - name: Attempt the rebase
+        id: attempt_the_rebase
+        run: |
+          git config --global user.email "julia-engineering@relational.ai"
+          git config --global user.name "RAI CI (GitHub Action Automation)"
+
+          git remote add upstream https://github.com/JuliaLang/julia
+          git fetch --quiet upstream ${{ matrix.julia_branch.upstream }}}
+          git rebase upstream/${{ matrix.julia_branch.upstream }}
+        continue-on-error: true
+      - name: On rebase success, push
+        if: steps.attempt_the_rebase.outcome == 'success'
+        run: |
+          git push --force origin ${{ matrix.julia_branch.name }} # force-push, because of the rebase.
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # --- Install Julia: Copied from CompatHelper ------------------------
+      - name: On rebase failure, install Julia nightly
+        if: steps.attempt_the_rebase.outcome != 'success'
+        uses: julia-actions/setup-julia@v1
+        with:
+          version: 'nightly'  # use the nightly build to do the Pkg.update()
+          arch: ${{ runner.arch }}
+      # --------------------------------------------------------------------
```

Comment:
we don't need this, right? in the raicode one, this is because we're going to attempt to resolve the Manifest.toml with nightly-Julia, but we're not doing that here
```suggestion
```